bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -495,6 +504,7 @@ public void run() {
 
                     journalStats.getForceWriteQueueSize().addCount(-requestsCount);
 
+                    Set<BookieRequestHandler> writeHandlers = new HashSet<>();
```

Comment:
Using the set has its own overhead too. In this case it will allocate a `Node` object for the linked list for each entry added here.

Reply:
Perhaps we could use `ObjectHashSet` from carrotsearch https://github.com/carrotsearch/hppc
It's already used in Pulsar too. 

We should also reuse the same set each time.